src/Visitation-Counters.ino

```diff
@@ -295,6 +301,7 @@ void loop()
       sysStatus.connectedStatus = true;
       sysStatus.lastConnection = Time.now();
       sysStatus.lastConnectionDuration = (millis()-connectionStartTime)/1000;   // How long - in seconds - did it take to connect
+      if (sysStatus.lastConnectionDuration > connectionTimeout) sysStatus.lastConnectionDuration = 0;
```

Comment:
Can you walk me through the logic here? If I'm reading it correctly, this will publish a `lastConnectionDuration` of 0 if it's larger than the `connectionTimeout`. Are we likely to end up here? The `else if` section of `if` statement will trigger if we have exceeded the `connectionTimeout`. I can see how we may end up here in an edge case - where we exceed the `connectionTimeout` during the loop, but connect to Particle cloud before returning to this `if` statement. In that case, we would only barely exceed `connectionTimeout` - do you want to 0 out these values in this edge case?

Reply:
so, when I updated the devices with the new code, it updated the sysStatus memory map to include a new INT value - lastConnectioDuration.  Until the device disconnected and reconnected from the Particle network, that value would be ~65k.  This value would be reported in the Webhook and recorded at Ubidots where it would totally screw up the virtualization of connection duration.  This was simply a way of testing for a number that would be out of bounds and setting it to zero where has less of an impact than 65k.  Now, this is just a "first connection" issue so perhaps there is a better way to deal with it.  At some point, perhaps I need to start writing back-end code that can clean up the Ubidots data and then I don't have to solve all my data issues on the device.  

Does this make sense?

Reply:
I think it does. I also think it is find to perform this check here. However, if I understand correctly, setting `lastConnectionDuration` to be 0 in the `setup()` function (which is done during loading system defaults and checking system values) should fix this issue, right?